internal/querynodev2/segments/segment_loader.go

```diff
@@ -1209,10 +1217,18 @@ func getResourceUsageEstimateOfSegment(schema *schemapb.CollectionSchema, loadIn
 			} else {
```

Comment:
Don't really understand the calcualtion logic here:

			indexMemorySize += neededMemSize -> if mmap enabled, do we still need same amount of memory to load it?
			if mmapEnabled {
				segmentDiskSize += neededMemSize + neededDiskSize
			} else {
				segmentDiskSize += neededDiskSize
			}

Reply:
please sync with @sunby and @liliu-z 
1. I'm assuming with latest approach, we don't need that much memory factor for loading index
2. local storage could greatly reduce the memory usage of load scalar data?
3. for mmap, we should see much lesser mamory usage compared to purely in memory?